### PR TITLE
Limit reference prefetch to Pipeline level, remove CRDS header caching

### PR DESCRIPTION
**Pull request by jaytmiller**
1. Refactor Pipeline and Step to move prefetch to Pipeline only

Moved reference file precaching from Step to Pipeline since precaching in Step
is of limited use and adds an extra CRDS operation and potential datamodel read
for each Step.  This leaves the Pipeline level prefetch and individual calls to
get_reference_file() within each Step.   The Pipeline level prefetch is essentially 
one operation for all Steps and continues to serve the purpose of broadly checking 
for errors and and caching references at the start of a Pipeline.

2. Disable the (dangerous) caching in the crds_client module.

Commented out active areas of CRDS client header + results caching effectively
disabling caching in crds_client w/o fully removing the code or refactoring
much.

### CODE REVIEW

File: jwst/assign_wcs/tests/test_nirspec.py [hunk 1]
```diff
@@ -366,6 +366,7 @@ def test_shutter_state(open_shutters, main_shutter, result):
     assert shutter_state == result
 
 
+@pytest.mark.xfail(reason="Corrected reference after CRDS prefetch change producing wrong slit count.")
```
Comment (nden) [inline jwst/assign_wcs/tests/test_nirspec.py]: 👍

Comment (jaytmiller) [reply jwst/assign_wcs/tests/test_nirspec.py]: Thanks.   I will note that I cannot merge this,  so if there is no objection,  someone please merge this in now.   (Fingers crossed,  it should at least avoid future incorrect assignments like this one.)

Comment (nden) [reply jwst/assign_wcs/tests/test_nirspec.py]: You should be able to merge now.

Comment (jaytmiller) [reply jwst/assign_wcs/tests/test_nirspec.py]: Nadia added me to maintainers,  by the power vested in me as CRDS-dude,  I'm merging,  fingers crossed,  unless someone else does first.

**Comment by hbushouse** [comment]
Fixes #1661

**Pull request merged**